src/pages/_app.tsx

```diff
@@ -147,168 +147,195 @@ class NRNApp extends NextApp {
    * @return {JSX.Element}
    */
   render(): JSX.Element {
-    const {
-      Component,
-      pageProps,
-      apollo,
-      router,
-      err,
-    }: AppRenderProps = this.props;
-
-    Sentry.configureScope((scope) => { // See https://www.npmjs.com/package/@sentry/node
-      // Only track meaningful data from router, as it contains lots of noise
-      scope.setContext('router', {
-        route: router.route,
-        pathname: router.pathname,
-        query: router.query,
-        asPath: router.asPath,
-      });
-    });
-
-    Sentry.addBreadcrumb({ // See https://docs.sentry.io/enriching-error-data/breadcrumbs
-      category: fileLabel,
-      message: `Rendering app for Component "${get(Component, 'name', 'unknown')}" (${isBrowser() ? 'browser' : 'server'})`,
-      level: Sentry.Severity.Debug,
-    });
-
-    const i18nextInstance = i18nextLocize(pageProps.lang, pageProps.defaultLocales); // Apply i18next configuration with Locize backend
+    const { Component, pageProps, apollo, err }: AppRenderProps = this.props;
 
-    // Build initial layout properties, they may be enhanced later on depending on the runtime engine
-    const layoutProps: LayoutProps = {
-      ...pageProps,
-      err,
-      router,
-      i18nextInstance,
-    };
+    if(isBrowser()){ // Avoids log clutter on server
+      console.debug('_app.render.pageProps', pageProps);
+    }
 
-    // XXX For an unknown reason, we noticed 2 render() calls. (each render call starts a new graphql request, and it makes debugging harder)
-    //  The first one doesn't contain any data from the server (no data, almost nothing) and therefore result in errors along the react sub tree
-    //  The second contains the expected data
-    //  Due to this behaviour, an "isSSRReadyToRender" variable has been introduced, to make sure we only render the components when all the data have been provided
-    if (layoutProps.isSSRReadyToRender) {
-      /**
-       * App rendered both on client or server (universal/isomorphic)
-       *
-       * @return {JSX.Element}
-       * @constructor
-       */
-      const UniversalApp = (): JSX.Element => (
+    if (pageProps.isReadyToRender || pageProps.isSSRReadyToRender || pageProps.statusCode === 404) {
```

Comment:
This `isReadyToRender` and `isSSRReadyToRender ` is weird, they shouldn't be necessary, if you do need it, it sounds like a bug somewhere else. Maybe `ApolloProvider` is forcing unnecessary renders, have you tried move it to an HOC, maybe `PageLayout ` and see how things play out?

Reply:
It's not related to ApolloProvider, I had this issue with a very basic example before using Apollo. Current way isn't clean but I haven't cleaned up the SSR approach yet, so it's more of a monkeypatch here.

Eventually, I believe a single isReadyToRender will be necessary, and will be set either from SSR or SSG.